paystakk/tests.py

```diff
@@ -1,100 +1,119 @@
-		api.create_invoice(customer=self.customer_code, amount=500000, due_date='2020-12-20')
+    def test_create_invoice(self):
+        api = Invoice(secret_key=SECRET_KEY, public_key=PUBLIC_KEY)
+        api.create_invoice(customer=self.customer_code,
+                           amount=500000, due_date='2020-12-20')
 
-		self.assertEqual(self.api.ctx.status, True)
+        self.assertEqual(self.api.ctx.status, True)
 
-	def test_list_invoices(self):
-		api = Invoice(secret_key=SECRET_KEY, public_key=PUBLIC_KEY)
-		api.list_invoices()
+    def test_list_invoices(self):
+        api = Invoice(secret_key=SECRET_KEY, public_key=PUBLIC_KEY)
+        api.list_invoices()
 
-		self.assertEqual(api.ctx.status, True)
+        self.assertEqual(api.ctx.status, True)
 
 
 class TestTransferControl(TestCase):
-	def test_balance(self):
-		api = TransferControl(secret_key=SECRET_KEY, public_key=PUBLIC_KEY)
-		api.get_balance()
+    def test_balance(self):
+        api = TransferControl(secret_key=SECRET_KEY, public_key=PUBLIC_KEY)
+        api.get_balance()
 
-		self.assertEqual(api.ctx.status, True)
-		self.assertEqual(api.ctx.message, 'Balances retrieved')
+        self.assertEqual(api.ctx.status, True)
+        self.assertEqual(api.ctx.message, 'Balances retrieved')
 
 
 class TestPaymentPage(TestCase):
-	def test_create_page(self):
-		r = PaymentPage(secret_key=SECRET_KEY, public_key=PUBLIC_KEY)
-		slug = str(time.time())
-		r.create_page(name='test page', slug=slug)
+    def test_create_page(self):
+        r = PaymentPage(secret_key=SECRET_KEY, public_key=PUBLIC_KEY)
+        slug = str(time.time())
+        r.create_page(name='test page', slug=slug)
+
+        self.assertEqual(r.ctx.status, True)
+        self.assertEqual(r.ctx.message, 'Page created')
+        self.assertEqual(r.name, 'test page')
+        self.assertEqual(r.slug, slug)
+        self.assertEqual(
+            r.page_url, 'https://paystack.com/pay/{slug}'.format(slug=slug))
 
-		self.assertEqual(r.ctx.status, True)
-		self.assertEqual(r.ctx.message, 'Page created')
-		self.assertEqual(r.name, 'test page')
-		self.assertEqual(r.slug, slug)
-		self.assertEqual(r.page_url, 'https://paystack.com/pay/{slug}'.format(slug=slug))
 
 class TestTransaction(TestCase):
-	def setUp(self):
-		self.api = Transaction(secret_key=SECRET_KEY, public_key=PUBLIC_KEY)
-		self.api.initialize_transaction(amount=30000,email='test@gmail.com')
-		# self.transaction_reference = self.api.transaction_reference
+    def setUp(self):
+        """
+        while testing the reference must be changed because each transaction
+        must have a unique reference,also there must be a predefined plan
+        before it can be used in the test.
+        """
+
+        self.api = Transaction(secret_key=SECRET_KEY, public_key=PUBLIC_KEY)
+        self.api.initialize_transaction(
+            amount=30000, email='test@gmail.com', reference="sales245",
+            invoice_limit=4, metadata={"custom_fields": [{"display_name":
```

Comment:
instead of writing the `metadata` part in that long form, why no do something like:
```(python)
self.api = Transaction(secret_key=SECRET_KEY, public_key=PUBLIC_KEY)
custom_fields = [{'display_name': 'Cart ID', ...}]
self.api.initialize_transaction(
            amount=30000, email='test@gmail.com', reference="sales245",
            invoice_limit=4, metadata={"custom_fields": custom_fields}...

```